constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -210,21 +210,29 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 		// TODO: should namespace be made available, if possible? Generally that context should be present
 		response := validator.Validate(ctx, versionedAttr.GetResource(), versionedAttr, constraint, nil, celAPI.PerCallLimit, nil)
 
-		enforcementAction, found, err := unstructured.NestedString(constraint.Object, "spec", "enforcementAction")
+		enforcementAction, err := apiconstraints.GetEnforcementAction(constraint)
```

Comment:
Query() should not care about enforcement action. All enforcement-action-processing-logic should live in the client.

Client logic would look like:

* Perform matching as usual and also check to make sure there is an associated enforcement action in order to presume a match (I think you already do this)

* Send a list of constraints-to-be-evaluated to the driver. The driver does not need to know what enforcement points are involved. It does not need to return any enforcement action. The driver can assume that if it has been passed a constraint, that constraint can/should be evaluated.

* Once the client receives results back from the driver, it can re-associate the responses with the enforcement actions that should be taken.

How enforcement actions should be returned from the client is a bit unknown... if we want to support parameters that customize the enforcement action, using a flat list might not work, but the CF interface is still a prototype, so not too worried there. In any case, that still doesn't require the driver to know about EAs/EPs.